### ISSUE TITLE
cli: could not find any subscriptions

### ISSUE THREAD

**Issue by vpukhanov** [MicrosoftDocs/vscodespaces#775]
I am unable to self-host a Codespace using the `codespaces` CLI tool because it is unable to find the Azure Subscription.

```
puhanov@ntp-docker-test:~/dev/codespace-root$ codespaces start --service
[2020-08-28 06:49:32.561 CLI.start I] Authenticating...
To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code D83ZSWKLF to authenticate.
[2020-08-28 06:49:49.922 CLI.start I] When connecting to this Codespace, you will be logged in as: puhanov
[2020-08-28 06:49:50.678 CLI.start E] [!ERROR] We could not find any subscriptions. Please create an Azure Subscription on https://azure.microsoft.com.
```

I have an active Azure account with a single user, a single Active Directory (the default one) and a single Pay-as-you-go subscription. Creating a Codespace through the web interface at https://online.visualstudio.com works as expected. Some additional system information:

```
puhanov@ntp-docker-test:~/dev/codespace-root$ codespaces -v
Command-line interface for Visual Studio Codespaces
1.0.2705

puhanov@ntp-docker-test:~/dev/codespace-root$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal
```

I encounter the same error on my personal PC, running the CLI tool gets me the same message. The setup works fine through the VS Code extension, but I don't have access to the desktop environment on the target server, so I can't use it there.

**Comment by OneCyrus** [comment]
same issue here. manually specifying the subscription by cli doesn't work too.

**Comment by vpukhanov** [comment]
Yep, same. Even specifying the fully qualified plan id gets me the same message

**Comment by taiyosogawa** [comment]
Taking a look. I've been able to repro with a personal account but it looks like work/school accounts are not affected.

**Comment by bchuecos** [comment]
We believe this might be an issue with a service upstream. We're filing an issue and will keep you updated

**Comment by janus-reith** [comment]
> Taking a look. I've been able to repro with a personal account but it looks like work/school accounts are not affected.

I wonder if that is the case. Previously, I could work around the issue described in #720 by setting up an Azure user as described here: https://github.com/MicrosoftDocs/vscodespaces/issues/720#issuecomment-670981082
But now, I am unable to register an instance with both my main acount and that azure account.
Could certainly be the case though that there's other differences between work accounts and the azure account which wouldn't be relevant for that other issue but are for this one.

**Comment by taiyosogawa** [comment]
@janus-reith, thanks for letting us know that it's not working with either of your accounts - that helps us understand the issue a little better. We're still checking in to see why our auth is not being recognized.

**Comment by jamesthurley** [comment]
I'm seeing the same issue. I created an Ubuntu 16.04 Azure VM and tried to run the Codespaces CLI tool, but it says "Could not find any subscriptions", whether I use the `--subscription` argument or the `--plan-id` argument. The Azure CLI authenticates successfully. Let me know if there is any additional information I can supply.

**Comment by vpukhanov** [comment]
Can a contributor clarify, how does the upcoming consolidation of VS Codespaces into GitHub Codespaces affect this issue? In my understanding, support for self-hosted codespaces under the GitHub service is [not currently planned](https://docs.microsoft.com/en-us/visualstudio/codespaces/resources/consolidation-faq#self-hosted-codespaces).

**Comment by janus-reith** [comment]
> Can a contributor clarify, how does the upcoming consolidation of VS Codespaces into GitHub Codespaces affect this issue? In my understanding, support for self-hosted codespaces under the GitHub service is [not currently planned](https://docs.microsoft.com/en-us/visualstudio/codespaces/resources/consolidation-faq#self-hosted-codespaces).

I really hope this functionality won't go away. I tried Github Codespaces provided Workspaces, and while I think it is really cool, it won't work out for my usecase in its current form - But the other way around, having my own instances integrated into Github would be cool, but not fundamentally relevant for me.

**Comment by bchuecos** [comment]
> Can a contributor clarify, how does the upcoming consolidation of VS Codespaces into GitHub Codespaces affect this issue? In my understanding, support for self-hosted codespaces under the GitHub service is [not currently planned](https://docs.microsoft.com/en-us/visualstudio/codespaces/resources/consolidation-faq#self-hosted-codespaces).

@nikmd23

**Comment by cheesecow** [comment]
https://docs.microsoft.com/en-us/visualstudio/codespaces/resources/consolidation-faq#self-hosted-codespaces
> Self-hosted is not currently included in the GitHub service, but we will be listening to feedback to inform our roadmap.

My company uses Github Enterprise and I didn't see any mention of this service being available for it. In addition we already have our own necessary development instances set up which due to security concerns won't be moved out of our managed hosting. It would be a shame to lose the self-hosted option for codespaces and would mean we would probably have to re-evaluate using VSCode.

Is it worth setting up another thread where people can thumbs-up or otherwise express interest in keeping self-hosted codespaces available?

**Comment by lostintangent** [comment]
@cheesecow That's a great suggestion! The discussion for GitHub Codespaces is happening in the [GitHub Support Community](https://github.community/c/codespaces-beta/45), and so if you want to start a thread there about self-hosted, we can carry on this conversation there 👍

**Comment by jamesthurley** [comment]
If self-hosted codespaces are not being ported across to GitHub Codespaces, does this mean that this issue is unlikely to be resolved?

I've started a discussion on the GitHub Support Community about supporting self-hosted codespaces here:

https://github.community/t/self-hosted-codespaces/131361

**Comment by AlexanderYW** [comment]
Having this issue as well on Ubuntu 20, tried specifying subscription as well, but with no luck, I hope this will be solved soon!

**Comment by cheesecow** [comment]
@AlexanderYW maybe you already did but some upvotes/a post on https://github.community/t/self-hosted-codespaces/131361 couldn't hurt 😉

**Comment by jacobdanovitch** [comment]
Having the same issue on Ubuntu 20.04.

**Comment by spanishgum** [comment]
I really hope this feature stays. I've been using the remote containers extension, but the docker experience for mac and windows has been pretty bad for me so far.

Being able to access a vscode instance on a Linux machine via the browser, and then launch containers with the remote containers extension, would be absolutely amazing.

**Comment by Guche20** [comment]
Good day. I have a similar challenge. I was able to create a resource group in Azure. However I still get errors and I am unable to create or use existing resource group to create a Code Space in Visual Studio Codespaces.

> @janus-reith, thanks for letting us know that it's not working with either of your accounts - that helps us understand the issue a little better. We're still checking in to see why our auth is not being recognized.

**Comment by subhankars** [comment]
> Good day. I have a similar challenge. I was able to create a resource group in Azure. However I still get errors and I am unable to create or use existing resource group to create a Code Space in Visual Studio Codespaces.
> 
> > @janus-reith, thanks for letting us know that it's not working with either of your accounts - that helps us understand the issue a little better. We're still checking in to see why our auth is not being recognized.

Same issue here!! it's frustrating :(
